tests/test_views.py

```diff
@@ -214,3 +224,243 @@ def test_select2_poll_content_view_pk(self):
+            mock.return_value = {PollContent: []}
+            results = self.view.get_data()
+
+        self.assertEqual(results.count(), 2)
+        self.assertIn(poll1, results)
+        self.assertIn(poll2, results)
+
+    def test_results_filtered_by_search_fields(self):
+        """
+        Check that if the search fields has been declared to filter against, the queryset is returned filtered
+        """
+        poll_content_contenttype_id = ContentType.objects.get_for_model(PollContent).id
+        poll = Poll.objects.create(name="Test poll")
+        poll1 = PollContent.objects.create(poll=poll, language="en", text="poll1")
+        poll2 = PollContent.objects.create(poll=poll, language="en", text="poll2")
+        self.request.GET = {"content_type_id": poll_content_contenttype_id, "query": "poll1"}
+
+        with patch("djangocms_navigation.views.supported_models") as mock:
+            mock.return_value = {PollContent: ["text"]}
+            results = self.view.get_data()
+
+        self.assertEqual(results.count(), 1)
+        self.assertIn(poll1, results)
+        self.assertNotIn(poll2, results)
+
+    def test_page_queryset_is_filtered_by_content_title(self):
+        """
+        Check that the returned Page QuerySet is filtered by the PageContent title
+        """
+        expected = PageContentFactory(title="Example PageContent", language="en")
+        PageContentFactory.create_batch(10)
+        self.request.GET = {"content_type_id": self.page_contenttype_id, "query": expected.title}
+
+        results = self.view.get_data()
+
+        self.assertEqual(Page._base_manager.count(), 11)
+        self.assertEqual(results.count(), 1)
+        self.assertIn(expected.page, results)
+
+    def test_page_queryset_filters_pages_by_current_language(self):
+        """
+        Check that only the page for the language of the request is returned
+        """
+        self.en = PageContentFactory(title="english test", language="en")
+        self.fr = PageContentFactory(title="french test", language="fr")
+        self.de = PageContentFactory(title="german test", language="de")
+        self.it = PageContentFactory(title="italian test", language="it")
+
+        for language in ["en", "fr", "de", "it"]:
+            with self.subTest(msg=language):
+                request = self.get_request(language=language)
+                request.GET = {"content_type_id": self.page_contenttype_id, "query": "test"}
+                self.view.request = request
+                results = self.view.get_data()
+
+                expected = getattr(self, language)
+                self.assertEqual(Page._base_manager.count(), 4)
+                self.assertEqual(results.count(), 1)
+                self.assertIn(expected.page, results)
+
+    @patch("djangocms_navigation.views.get_current_site")
+    def test_page_queryset_filtered_by_site(self, mock_get_current_site):
+        """
+        Check that when pages belong to different sites, only sites for
```

Comment:
Not sure if this is docstring is complete?

Reply:
Oops, fixed